tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/LinagoraEncryptedEmailDetailedViewGetMethodContract.scala

```diff
@@ -0,0 +1,1003 @@
+           |    ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def methodShouldReturnNotFoundWhenEncryptedEmailContentDoesNotExist(server: GuiceJamesServer): Unit = {
+    val messageId: MessageId = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString(),
+        BOB_INBOX_PATH,
+        AppendCommand.from(MESSAGE))
+      .getMessageId
+
+    server.getProbe(classOf[JmapGuiceEncryptedEmailContentStoreProbe])
+      .delete(messageId)
+
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:pgp"],
+         |  "methodCalls": [
+         |    ["EncryptedEmailDetailedView/get", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "ids": ["${messageId.serialize}"]
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .isEqualTo(
+        s"""{
+           |    "sessionState": "${SESSION_STATE.value}",
+           |    "methodResponses": [
+           |        [
+           |            "EncryptedEmailDetailedView/get",
+           |            {
+           |                "accountId": "$ACCOUNT_ID",
+           |                "state": "${INSTANCE.value}",
+           |                "notFound": [
+           |                    "${messageId.serialize}"
+           |                ]
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def methodShouldReturnDetailedViewWhenEmailIdExits(server: GuiceJamesServer): Unit = {
+    val messageId: MessageId = server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString(),
+        BOB_INBOX_PATH,
+        AppendCommand.from(MESSAGE))
+      .getMessageId
+    val request: String =
+      s"""{
+         |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:pgp"],
+         |  "methodCalls": [
+         |    ["EncryptedEmailDetailedView/get", {
+         |      "accountId": "$ACCOUNT_ID",
+         |      "ids": ["${messageId.serialize}"]
+         |    }, "c1"]
+         |  ]
+         |}""".stripMargin
+
+    val response: String = `given`
+      .body(request)
+    .when()
+      .post()
+    .`then`
+      .statusCode(HttpStatus.SC_OK)
+      .contentType(JSON)
+      .extract()
+      .body()
+      .asString()
+
+    assertThatJson(response)
+      .inPath("methodResponses[0][1]")
+      .isEqualTo(
+        s"""{
+           |    "accountId": "$ACCOUNT_ID",
+           |    "state": "${INSTANCE.value}",
+           |    "list": [
+           |        [
+           |            "${messageId.serialize()}",
+           |            {
+           |                "encryptedPreview": "$${json-unit.ignore}",
```

Comment:
Convenient trick :+1:  <3

Reply:
That just ignore the value here? Yeah that's awesome, good finding :)